### ISSUE TITLE
[BUG][Kotlin]Description

### ISSUE THREAD

**Issue by crazyk2** [OpenAPITools/openapi-generator#5492]
@jimschubert @dr4ke616 @karismann @Zomzog @andrewemery @4brunu
What result shoud be when you pass var name `111` or `111name`?
By according [this](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java#L869) comment I think it should be escaped `` `111` `` and `` `111name` `` but [this class](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java#L42) and [method](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java#L835) are full of bugs and result is `toVarName()` is `_111` and  `` `111name` ``. As for me it's different strange behavior. As for Kotlin it's acceptable. But I want to make some refactor of this class and I need approve of result.

**Comment by chameleon82** [comment]
Scala templates have similar problem. See https://github.com/OpenAPITools/openapi-generator/issues/5496 and related PR
Generally, all templates of particular language should have same variable styles.
Basically, all cases should be covered with [tests](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/java/org/openapitools/codegen/kotlin/AbstractKotlinCodegenTest.java#L127) regarding to language naming convention. All specific cases should have some explanation. For example, it is not clear, why `UPPERCASE` variable should not be camelized as there is no specific issue for that.

**Comment by crazyk2** [comment]
@chameleon82 I agree with you, but language allows you use different styles for different tastes. And now I see a mix without a clear separation of styles. And I ask community about their minds/ because  I already did this job locally.

**Comment by chameleon82** [comment]
@crazyk2 generally it is a bad practice to not following language name convention, especially in community projects. But I agree, in some corner cases need to break this rule. So, imho, those corner cases should be clear.
I definetly agree we can clean it up as we already did it for scala. Could you open PR against AbstractKotlinGenerator so we can apply basic rules for all kotlin templates?
